File: backend/PxlTeambuilderApi/PxlTeambuilderApi.Data/Domain/Group.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace PxlTeambuilderApi.Data.Domain
{
    public class Group
    {
        private ICollection<User> teamMembers = new List<User>();

        public string GroupId { get; set; }
        public string Name { get; set; }
        public string ProjectId { get; set; }

        //navigation properties
        [JsonIgnore]
        public virtual ICollection<UserProjectDetail> UserProjectDetails { get; set; }

        [JsonIgnore]
        public virtual Project Project { get; set; }

        //non database properties
        public ICollection<User> TeamMembers
        {
            get
            {
                return this.teamMembers;
            }

            set
            {
                this.teamMembers = value;
            }
        }
    }
}

File: frontend/pxlTeambuilder/src/app/models/group.ts
import { User } from './user';

export class Group{
    
    private id : string;
    private name : string;
    private projectId : string;
    private teammembers : User[]


    constructor(){}

    public get Id(){
        return this.id;
    }

    public get Name(){
        return this.name;
    }

    public get ProjectId(){
        return this.projectId;
    }

    public get TeamMembers(){
        return this.teammembers;
    }

    public set Id(id : string){
        this.id = id;
    }

    public set Name(name : string){
        this.name = name;
    }

    public set ProjectId(projectId : string){
        this.projectId = projectId;
    }

    public set TeamMembers(teammembers : User[]){
        this.teammembers = teammembers;
    }
}